x/emissions/keeper/msgserver/msg_server_worker_payload.go

```diff
@@ -22,15 +22,18 @@ func (ms msgServer) VerifyAndInsertInferencesFromTopInferers(
 	workerDataBundles []*types.WorkerDataBundle,
 	maxTopWorkersToReward uint64,
 ) (map[string]bool, error) {
-
 	inferencesByInferer := make(map[string]*types.Inference)
 	latestInfererScores := make(map[string]types.Score)
 	for _, workerDataBundle := range workerDataBundles {
 		/// Do filters first, then consider the inferenes for inclusion
 		/// Do filters on the per payload first, then on each inferer
 		/// All filters should be done in order of increasing computational complexity
 
-		// check signatures from the bundle throw if invalid!
+		if err := workerDataBundle.Validate(); err != nil {
+			return nil, err
+		}
+
+		// Check signatures from the bundle throw if invalid!
 		pk, err := hex.DecodeString(workerDataBundle.Pubkey)
 		if err != nil || len(pk) != secp256k1.PubKeySize {
 			return nil, types.ErrSignatureVerificationFailed
```

Comment:
Here, if one bundle is invalid, it will throw the whole set as invalid (`return`,  not `continue` ). 
I think only that bundle should fail.
Other bundles should not have to suffer from one bundle being signed incorrectly.

Unless what we want here is to leave the verification also on the node leader who makes the insertion - in which case we'd need to make that explicit, and if it is not verified, punish the leader somehow for not verifying the data sent.

Applies to any `return`s in this function, there are other ones in the loop.